src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java

```diff
@@ -171,68 +172,51 @@ private void hideDetails(@RequestBody TextSubmission textSubmission) {
     }
 
     /**
-     * GET  /text-submissions : get all the textSubmissions.
+     * GET  /text-submissions : get all the textSubmissions for an exercise. It is possible to filter, to receive only
+     * the one that have been already submitted, or only the one assessed by the tutor who is doing the call
      *
      * @return the ResponseEntity with status 200 (OK) and the list of textSubmissions in body
      */
     @GetMapping(value = "/exercises/{exerciseId}/text-submissions")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Transactional(readOnly = true)
     public ResponseEntity<List<TextSubmission>> getAllTextSubmissions(@PathVariable Long exerciseId,
-                                                                      @RequestParam(defaultValue = "false") boolean submittedOnly) {
+                                                                      @RequestParam(defaultValue = "false") boolean submittedOnly,
+                                                                      @RequestParam(defaultValue = "false") boolean assessedByTutor) {
         log.debug("REST request to get all TextSubmissions");
         Exercise exercise = exerciseService.findOneLoadParticipations(exerciseId);
```

Comment:
You don't need to load the exercise with all participations here as you load the participations in a different call later on. So please only use findOne(exerciseId) here

Reply:
Done